core/src/test/java/io/hyperfoil/core/impl/statistics/StatisticsStoreTest.java

```diff
@@ -77,39 +76,101 @@ public void allJson_empty_issue45() {
+            .endPhase()
+
+            .addPhase("ramp/001/two").always(1)
+            .duration(60_000)
+            .scenario().initialSequence("two")
+            .step(StepCatalog.class).httpRequest(HttpMethod.GET)
+            .sla().addItem().meanResponseTime(1, TimeUnit.MILLISECONDS).endSLA().endList()
+            .endStep()
+            .endSequence().endScenario()
+            .endPhase()
+
+            .addPhase("ramp/002/one").always(1)
             .duration(60_000)
             .scenario().initialSequence("test")
             .step(StepCatalog.class).httpRequest(HttpMethod.GET)
             .sla().addItem().meanResponseTime(1, TimeUnit.MILLISECONDS).endSLA().endList()
             .endStep()
             .endSequence().endScenario()
             .endPhase()
+            .addPhase("steady/002/one").always(1)
+            .duration(60_000)
+            .scenario().initialSequence("test")
+            .step(StepCatalog.class).httpRequest(HttpMethod.GET)
+            .sla().addItem().meanResponseTime(1, TimeUnit.MILLISECONDS).endSLA().endList()
+            .endStep()
+            .endSequence().endScenario()
+            .endPhase()
+
+
             .build();
-      Phase phase = benchmark.phases().stream().findAny().get();
-      HttpRequestStep step = (HttpRequestStep) Stream.of(phase.scenario().sequences())
+      Iterator<Phase> phaseIterator = benchmark.phases().iterator();
+      Phase phase1 = phaseIterator.next();
+      Phase phase2 = phaseIterator.next();
+      Phase phase3 = phaseIterator.next();
+      Phase phase4 = phaseIterator.next();
+
+      HttpRequestStep step1 = (HttpRequestStep) Stream.of(phase1.scenario().sequences())
+            .flatMap(s -> Stream.of(s.steps()))
+            .filter(HttpRequestStep.class::isInstance)
+            .findAny().get();
+      HttpRequestStep step2 = (HttpRequestStep) Stream.of(phase2.scenario().sequences())
+            .flatMap(s -> Stream.of(s.steps()))
+            .filter(HttpRequestStep.class::isInstance)
+            .findAny().get();
+      HttpRequestStep step3 = (HttpRequestStep) Stream.of(phase3.scenario().sequences())
+            .flatMap(s -> Stream.of(s.steps()))
+            .filter(HttpRequestStep.class::isInstance)
+            .findAny().get();
+      HttpRequestStep step4 = (HttpRequestStep) Stream.of(phase4.scenario().sequences())
             .flatMap(s -> Stream.of(s.steps()))
             .filter(HttpRequestStep.class::isInstance)
             .findAny().get();
 
       StatisticsStore store = new StatisticsStore(benchmark, failure -> {
       });
 
-      store.record("address", phase.id(), step.id(), "metric1",
+
+      store.record("address", phase1.id(), step1.id(), "metric1",
+            makeSnapshot(101, 0 * 60_000, 1 * 60_000, 1_000, 10_000, 30_000, 60_000, 120_000, 666_000_000)
+      );
+      store.recordSessionStats("address", 0 * 60_000, "ramp/001/one", 1, 10);
+      store.completePhase("ramp/001/one");
```

Comment:
I don't think that you can rely on benchmark.phases() returning the phases in any particular order.